### ISSUE TITLE
add to the list of default web browsers on Windows 10

### ISSUE THREAD

**Issue by englishextra** [Eloston/ungoogled-chromium#623]
This is to share. Not a question or bug.

The .reg file below will not set the Ungoogled Chromium as default browser, it only shows it in Control Panel > Default Programs > Set default programs

```ini
Windows Registry Editor Version 5.00

; Infamous capabilities:

[HKEY_LOCAL_MACHINE\SOFTWARE\ungoogled-chromium\Capabilities]
"ApplicationDescription"="ungoogled-chromium"
"ApplicationIcon"="C:\\Program Files\\ungoogled-chromium\\chrome.exe,0"
"ApplicationName"="ungoogled-chromium"

[HKEY_LOCAL_MACHINE\SOFTWARE\ungoogled-chromium\Capabilities\FileAssociations]
".htm"="ungoogled-chromiumURL"
".html"="ungoogled-chromiumURL"
".shtml"="ungoogled-chromiumURL"
".xht"="ungoogled-chromiumURL"
".xhtml"="ungoogled-chromiumURL"

[HKEY_LOCAL_MACHINE\SOFTWARE\ungoogled-chromium\Capabilities\URLAssociations]
"ftp"="ungoogled-chromiumURL"
"http"="ungoogled-chromiumURL"
"https"="ungoogled-chromiumURL"

; Register to Default Programs

[HKEY_LOCAL_MACHINE\SOFTWARE\RegisteredApplications]
"ungoogled-chromium"="Software\\ungoogled-chromium\\Capabilities"

; ungoogled-chromiumURL HANDLER:

[HKEY_LOCAL_MACHINE\Software\Classes\ungoogled-chromiumURL]
@="ungoogled-chromium Document"
"FriendlyTypeName"="ungoogled-chromium Document"

[HKEY_LOCAL_MACHINE\Software\Classes\ungoogled-chromiumURL\shell]

[HKEY_LOCAL_MACHINE\Software\Classes\ungoogled-chromiumURL\shell\open]

[HKEY_LOCAL_MACHINE\Software\Classes\ungoogled-chromiumURL\shell\open\command]
@="\"C:\\Program Files\\ungoogled-chromium\\chrome.exe\" \"%1\""
```

**Comment by Eloston** [comment]
Thanks for sharing! To reduce clutter, issues are not for submitting tips. Instead, please consider submitting a PR to the [ungoogled-chromium-wiki](https://ungoogled-software.github.io/ungoogled-chromium-wiki/).

**Comment by arturkomoter** [comment]
After running this script, Chromium still can't tell that it is the default browser. What to do?

**Comment by englishextra** [comment]
it wont tell you anything please read the issue in the first post

**Comment by xhiddle** [comment]
So there's this lovely freeware tool

https://kolbi.cz/blog/2017/11/10/setdefaultbrowser-set-the-default-browser-per-user-on-windows-10-and-server-2016-build-1607/

But because UNgoogled chromium prebuilt binary was a zip file it's not hooked into all the right places so more magic required

There is this other tool that references the first tool which explains that a hash must be generated to bind new associations prossibly explaining why OP does not result in satisfaction

https://kolbi.cz/blog/2017/10/25/setuserfta-userchoice-hash-defeated-set-file-type-associations-per-user/

which itself cryptically references GPO  (not for  win 10 home users).  Good thing I have pro?

**Comment by Innomen** [comment]
> So there's this lovely freeware tool
> 
> https://kolbi.cz/blog/2017/11/10/setdefaultbrowser-set-the-default-browser-per-user-on-windows-10-and-server-2016-build-1607/
> 
> But because UNgoogled chromium prebuilt binary was a zip file it's not hooked into all the right places so more magic required
> 
> There is this other tool that references the first tool which explains that a hash must be generated to bind new associations prossibly explaining why OP does not result in satisfaction
> 
> https://kolbi.cz/blog/2017/10/25/setuserfta-userchoice-hash-defeated-set-file-type-associations-per-user/
> 
> which itself cryptically references GPO (not for win 10 home users). Good thing I have pro?

Does nothing for me, window opens briefly and then closes, no changes.

**Comment by Neelsie** [comment]
Hello, 
Found a solution if none of these others are working.
I had an issue where window's default browser directory for chromium (portable version at chromium.woolyss.com) was at the downloads folder and windows had not detected the missing executable for the default browser, icon was missing too in w10 settings.

After trying many things I eventually got a fix, updating the directory values at

**`Computer\HKEY_CLASSES_ROOT\Chromium(randomstring)\shell\open\command`**

at "\HKEY_CLASSES_ROOT\Chromium(randomstring)\" the random letters/numbers at the end of chromium will be different for everyone im assuming so just look for chromium and you'll see it.

good luck
